lib/useSubscription.tsx

```diff
@@ -22,7 +22,7 @@ export default function useSubscription(
 
   const callback = useCallback(
     (receivedTopic: string, receivedMessage: any) => {
-      if ([topic].flat().some(rTopic => matches(topic, rTopic))) {
+      if ([receivedTopic].flat().some(rTopic => matches(topic, rTopic))) {
```

Comment:
I think the right thing would be:
`if ([topic].flat().some(rTopic => matches(rTopic, receivedTopic))) {`

Reply:
I agree with you，I made a mistake。